lib/ui/components/large_button_widget.dart

```diff
@@ -0,0 +1,313 @@
+        buttonType == ButtonType.critical ||
+        buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeBase;
+    }
+    return null;
+  }
+
+  //Returning null to fallback to default color
+  Color? _disabledBorderColor(EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.primary ||
+        buttonType == ButtonType.secondary ||
+        buttonType == ButtonType.critical) {
+      return Colors.transparent;
+    }
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeMuted;
+    }
+    return null;
+  }
+
+  Color _defaultIconColor({
+    required EnteColorScheme colorScheme,
+    required EnteColorScheme inverseColorScheme,
+  }) {
+    if (_isPrimary() || buttonType == ButtonType.critical) {
+      return strokeBaseDark;
+    }
+    if (_isSecondary()) {
+      return colorScheme.strokeBase;
+    }
+    if (buttonType == ButtonType.neutral ||
+        buttonType == ButtonType.trailingIcon) {
+      return inverseColorScheme.strokeBase;
+    }
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.warning500;
+    }
+    //fallback
+    return colorScheme.strokeBase;
+  }
+
+  //Returning null to fallback to default color
+  Color? _pressedIconColor(EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeBase;
+    }
+    return null;
+  }
+
+  //Returning null to fallback to default color
+  Color? _disabledIconColor(EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.primary ||
+        buttonType == ButtonType.secondary) {
+      return colorScheme.strokeMuted;
+    }
+    if (buttonType == ButtonType.critical ||
+        buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeFaint;
+    }
+    return null;
+  }
+
+  TextStyle _defaultLabelStyle({
+    required EnteTextTheme textTheme,
+    required EnteTextTheme inverseTextTheme,
+  }) {
+    if (_isPrimary() || buttonType == ButtonType.critical) {
+      return textTheme.bodyBold.copyWith(color: textBaseDark);
+    }
+    if (_isSecondary()) {
+      return textTheme.bodyBold;
+    }
+    if (buttonType == ButtonType.neutral ||
+        buttonType == ButtonType.trailingIcon) {
+      return inverseTextTheme.bodyBold;
+    }
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return textTheme.bodyBold.copyWith(color: warning500);
+    }
+    //fallback
+    return textTheme.bodyBold;
+  }
+
+  //Returning null to fallback to default color
+  TextStyle? _pressedLabelStyle(
+      EnteTextTheme textTheme, EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return textTheme.bodyBold.copyWith(color: colorScheme.strokeBase);
+    }
+    return null;
+  }
+
+  //Returning null to fallback to default color
+  TextStyle? _disabledLabelStyle(
+    EnteTextTheme textTheme,
+    EnteColorScheme colorScheme,
+  ) {
+    if (buttonType == ButtonType.primary ||
```

Comment:
eval if this also can be extension method inside `extension ExtentendButtonType on ButtonType ` (more below)

Reply:
Moved methods like these to inside the enum using enhanced enums.